lexicons/com/atproto/repo/importRepo.json

```diff
@@ -0,0 +1,13 @@
+{
+  "lexicon": 1,
+  "id": "com.atproto.repo.importRepo",
+  "defs": {
+    "main": {
+      "type": "procedure",
+      "description": "Import a repo in the form of a CAR file. Returns a 202 after successfully reading incoming CAR then asynchronously indexes the repo and contained records",
```

Comment:
I think we discussed requiring `Content-Length` HTTP header for this one; if so should mention in description.

Should probably also clarify when this endpoint can be called: only during migration? at any time, overwriting existing repo state? only when account is deactivated?

Reply:
Good call on the content-length header.

Hmm yeah I'm not quite sure on when it can be called 🤔 

Part of me thinks it'd be fine to call it at any time?

Reply:
It is a somewhat destructive operation (wipes existing content, I assume). It would be bad if somebody thought it was additive, imported a small CAR, and it blasted all their stuff away. I think it would blast all downstream copies fast also (eg, Relay would ~immediately have the new small copy). This concern could be mitigated by having a param like `clobberExisting` which makes it really clear you are doing something dangerous. It still feels like this is almost in the territory of needing to verify with email/token (like deleting account).

There are also some resource consumption issues: if you kept importing large but different repos, it would consume PDS, Relay, AppView, search, automod, etc resources every time. This can be mitigated by just tightly rate-limiting this endpoint.

New question: what kind of validation of the commit object is done, if any? I assume we should ignore signature (might be an old CAR file, with out-of-date signing key). What if the DID doesn't match? Probably fine in most cases.

How does the new PDS chose a `rev` for the new commit after import? If we lean in to `rev` as TID, it is easy: just use the current timestamp TID and everything should work fine. If the imported commit object has a DID match, and a `rev` number, could also start incrementing from that previous `rev`, though I think this is weaker (maybe the CAR export didn't include some more recent commits, and `rev` would not be current). (I of course lean towards using current TID for rev, but laying out the options)

Reply:
Good points on the destructiveness & possible abuse vector 🤔 I'll lock this down a bit so it can really only be used for account migration right now (we can make that more lenient in the future if we see the need)

For new commits, we just use the current TID as you pointed out 👍 

In terms of validation on commit object: We don't do any. we just import the "data" part of the repo. We ignore signature for the reason you mention. I don't bother checking did/rev because there's not really any attack you're preventing if we're not checking the signature 🤷‍♂️ 

In a way this is kind've a "massive bulk `applyWrites`" where the writes are just whatever is including in the imported MST